### ISSUE TITLE
CI release has failed

### ISSUE THREAD

**Issue by mockitoguy** [mockito/shipkit#512]
It seems that the CI job has failed: https://travis-ci.org/mockito/shipkit/builds/288214545

**Comment by epeee** [comment]
This one some seems to be related to #395.
The first failing [build](https://travis-ci.org/mockito/shipkit/builds/288214072) failed with the same error message as the one in the linked issue.

```
[performRelease] > Execution of command failed (exit code 1):
[performRelease]     git push https://dummy:[SECRET]@github.com/mockito/shipkit.git v0.9.115 master
[performRelease]     Captured command output:
[performRelease]   To https://github.com/mockito/shipkit.git
[performRelease]    * [new tag]           v0.9.115 -> v0.9.115
[performRelease]    ! [rejected]          master -> master (non-fast-forward)
[performRelease]   error: failed to push some refs to 'https://dummy:[SECRET]@github.com/mockito/shipkit.git'
[performRelease]   hint: Updates were rejected because the tip of your current branch is behind
[performRelease]   hint: its remote counterpart. Integrate the remote changes (e.g.
```

**Comment by mockitoguy** [comment]
I'm not sure why it is failing. I restarted the build and it failed with:

```
[performRelease] :fetchReleaseNotes FAILED
[performRelease] 
[performRelease] FAILURE: Build failed with an exception.
[performRelease] 
[performRelease] * What went wrong:
[performRelease] Execution failed for task ':fetchReleaseNotes'.
[performRelease] > Execution of command failed (exit code 128):
[performRelease]     git log --pretty=%ad --date=iso v0.9.115 -n 1
[performRelease]     Captured command output:
[performRelease]   fatal: ambiguous argument 'v0.9.115': unknown revision or path not in the working tree.
[performRelease]   Use '--' to separate paths from revisions, like this:
[performRelease]   'git <command> [<revision>...] -- [<file>...]'
```

However "git log --pretty=%ad --date=iso v0.9.115 -n 1" command works for me on clean checkout. Is Travis CI healthy? It looks like it does not see the tag but the tag does exist.

Can you take a look at it? I should have some time middle next week to look into this.

BTW. I created new ticket #513 to help debugging CI release failures.

**Comment by epeee** [comment]
Yes, I will have a look at it.

**Comment by epeee** [comment]
Luckily, I was able to reproduce it locally.
v0.9.115 tag was not available after updating master.
After calling "git fetch --tags" the missing tags appeared and I was no longer heading into the problem described above.
It looks like the tag (v0.9.115) is not set on master? In fact we do have two 0.9.115 commits (one of them on master, but the oder one seems to have the tag).

**Comment by epeee** [comment]
Travis uses
"git clone --depth=50 --branch=master https://github.com/mockito/shipkit.git mockito/shipkit"
After using this command the tag is also missing locally.

**Comment by epeee** [comment]
I'd suggest to go for a "git pull" right before the git push. It looks like this will solve most of our problems with pr's being merged right after each other. Any concerns about this solution?

**Comment by mockitoguy** [comment]
Thanks for investigating. Looking at it right now.

**Comment by mockitoguy** [comment]
This should fix the problem: #515
